### PR TITLE
fix bug - k8s was crashing when yaml ended with 3 dashes

### PR DESCRIPTION
**Pull request by michaszcz**
##### SUMMARY
This PR fixes following issue: 
Ansible was crashing when provided yaml ended with 3 dashes, for example this one (https://raw.githubusercontent.com/jetstack/cert-manager/release-0.9/deploy/manifests/00-crds.yaml)

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
k8s

### CODE REVIEW

**Review by fabianvf — APPROVED** [review]
shipit

File: lib/ansible/module_utils/k8s/common.py [hunk 1]
```diff
@@ -246,7 +246,7 @@ def load_resource_definitions(self, src):
             self.fail(msg="Error accessing {0}. Does the file exist?".format(path))
         try:
             with open(path, 'r') as f:
-                result = list(yaml.safe_load_all(f))
+                result = list(filter(None, yaml.safe_load_all(f)))
```
Comment (willthames) [inline lib/ansible/module_utils/k8s/common.py]: my understanding is that `filter` is less pythonic than list comprehensions:

`result = [item for item in yaml.safe_load_all(f) if item]`

I also struggled with the readability of the `filter` test case - I would naively expect that to return all elements that were `None`

Comment (michaszcz) [reply lib/ansible/module_utils/k8s/common.py]: I agree, new commit has been added.

**Comment by willthames** [comment]
@Shaid3r could you add a test for this functionality?

Looks like we already have a [test for multi-document YAML](https://github.com/ansible/ansible/blob/devel/test/integration/targets/k8s/tasks/full_test.yml#L206-L218), would be good to capture the case where it ends (or starts) with a `---`

**Comment by Shaid3r** [comment]
@willthames Done.

**Comment by willthames** [comment]
Sorry @Shaid3r, I remembered one more thing - could you add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to) for this change. With a changelog fragment it'll be easier to backport this to older versions, which I think would be wise (even if only current 2.8)

**Comment by willthames** [comment]
Once this has the changelog fragment this is good to merge (test suite passes fine)

**Comment by willthames** [comment]
Merged, thanks @Shaid3r !
